### PR TITLE
Add HubSpot integration for a few API endpoints dealing with contacts.

### PR DESCRIPTION
**Pull request by vrv**
Adds support for a few different contact APIs.

- List all your contacts.
- Create a new contact (for now, only supports creating with the basic properties - update contact is needed to update it with more properties.).
- Update a contact.
- Search for contacts.

### CODE REVIEW

File: hubspot/plugin.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+@dataclass
+class HubSpotContactProperties:
+    first_name: str
+    last_name: str
+    email: str
+    hs_object_id: str
+    last_modified_date: datetime
+
+    # These are additional properties that are available in the HubSpot contact system
```
Comment (bozhi) [inline hubspot/plugin.py:18]: Did you want this to appear as part of the doc_string so that code_gen has access to it?

Comment (vrv) [reply hubspot/plugin.py]: Yep, good point, thanks!

**Comment by vrv** [comment]
(will fix forward any fixes)

File: hubspot/plugin.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    # Construct the filters based on the search criteria
+    filters = []
+    for property_name, value in search_criteria.items():
+        filters.append(
+            {"propertyName": property_name, "operator": "EQ", "value": value}
+        )
+
+    properties = ["firstname", "lastname", "email", "lastmodifieddate"]
+    if additional_properties:
+        properties = list(set(properties + list(additional_properties)))
```
Comment (jcharum) [inline hubspot/plugin.py:218]:
```suggestion
        properties = list(set(*properties, *additional_properties))
```

File: hubspot/plugin.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        }
+        contacts_payload.append(contact_data)
+
+    payload = {"inputs": contacts_payload}
+
+    with httpx.Client(
+        transport=AugmentedTransport(actions_v0.authenticated_request_hubspot),
+    ) as client:
+        response = client.post(url, json=payload)
+        response.raise_for_status()  # Make sure to handle potential errors here
```
Comment (jcharum) [inline hubspot/plugin.py]:
```suggestion
        response.raise_for_status()
```
IMHO superfluous.

File: hubspot/plugin.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    with httpx.Client(
+        transport=AugmentedTransport(actions_v0.authenticated_request_hubspot),
+    ) as client:
+        response = client.post(url, json=payload)
+        response.raise_for_status()
+        data = response.json()
+
+    for item in data.get("results", []):
+        properties = item.get("properties", {})
+        addl_props = {}
```
Comment (jcharum) [inline hubspot/plugin.py:233]: Perhaps `property_values` and `additional_property_values` to avoid the "shadowing" and change in abbreviation?

**Review by jcharum — COMMENTED** [review]
It appears that this was merged as I was reviewing it.

**Comment by vrv** [comment]
> It appears that this was merged as I was reviewing it.

sorry, wanted to test something in prod! :)  sending another PR to fix these

File: hubspot/plugin.py (hunk 2 above)
Comment (jcharum) [reply hubspot/plugin.py:218]: Oops; maybe needs:
```python
properties = list(set([*properties, *additional_properties]))
```

Comment (vrv) [reply hubspot/plugin.py:218]: Ruff auto-fixed to 

`properties = list({*properties, *additional_properties})` :)

Comment (vrv) [reply hubspot/plugin.py:218]: I don't fully understand why set(*properties, *additional_properties) fails but {*properties, *additional_properties} works, but I am new to Python

**Pull request merged**